File: java/account/DateImpl.java
package account;

import java.util.Calendar;

import connection.Connection;
import corbaAccount.date;
import corbaAccount.datePOA;

public class DateImpl extends datePOA {

	private int _day;
	private int _month;
	private int _year;
	private Connection _connection;

	public DateImpl(date d) throws Exception {
		super();
		day(d.day());
		month(d.month());
		year(d.year());
		_connection = Connection.getInstance();
		_connection.activateServant(this);
	}
	
	public DateImpl(int year, int month, int day) throws Exception {
		super();
		day(day);
		month(month);
		year(year);
		_connection = Connection.getInstance();
		_connection.activateServant(this);
	}
	
	public DateImpl() throws Exception {
		this(Calendar.getInstance().get(Calendar.YEAR), Calendar.getInstance().get(Calendar.MONTH), Calendar.getInstance().get(Calendar.DAY_OF_MONTH));
	}
	
	@Override
	public int day() {
		return _day;
	}

	@Override
	public void day(int newDay) {
		_day = newDay;
	}

	@Override
	public int month() {
		return _month;
	}

	@Override
	public void month(int newMonth) {
		_month = newMonth;
	}

	@Override
	public int year() {
		return _year;
	}

	@Override
	public void year(int newYear) {
		_year = newYear;
	}

	@Override
	public String _toString() {
		return String.valueOf(_year) + "/" + 
				String.format("%2s",Integer.valueOf(_month)).replaceAll(" " , "0") + "/" +
				String.format("%2s",Integer.valueOf(_day)).replaceAll(" " , "0");
	}
	
}

File: cxx/account/DateImpl.cpp
/**
 * @file DateImpl.cpp
 * @author Ronald T. Fernandez
 * @version 1.0
 */

#include "DateImpl.h"

#include <exception>
#include <iostream>
#include <iomanip>
#include <sstream>

namespace account {

std::shared_ptr<connection::Connection> DateImpl::_connection = nullptr;

DateImpl::~DateImpl() {
	_connection->deactivateServant(this);
}

DateImpl::DateImpl(::corbaAccount::date_ptr d) {
	str = "";
	if (CORBA::is_nil(d)) {
		std::cerr << "Cannot set a NULL date" << std::endl;
		throw std::exception();
	}
	DateImpl(d->year(), d->month(), d->day());
}

DateImpl::DateImpl(int year, int month, int day) : _year(year), _month(month), _day(day) {
	std::ostringstream os;
	os << std::setw(2) << std::setfill('0') << this->_year << "/" << this->_month << "/" << this->_day;
	str = os.str();

	// Activate the CORBA object
	_connection = connection::Connection::getInstance();
	_connection->activateServant(this);
}

DateImpl::DateImpl() {
	DateImpl(0, 0, 0);
};

::CORBA::Long DateImpl::day() {
	return this->_day;
}
void DateImpl::day(::CORBA::Long _v) {
	this->_day = _v;
}
::CORBA::Long DateImpl::month() {
	return this->_month;
}
void DateImpl::month(::CORBA::Long _v) {
	this->_month = _v;
}
::CORBA::Long DateImpl::year() {
	return this->_year;
}
void DateImpl::year(::CORBA::Long _v) {
	this->_year = _v;
}

char* DateImpl::toString() {
	return CORBA::string_dup(str.c_str());
};

} /* namespace account */
